mirrord/layer/src/file/hooks.rs

```diff
@@ -133,6 +133,17 @@ pub(super) unsafe extern "C" fn open_nocancel_detour(
     }
 }
 
+/// Hook for [`libc::opendir`].
+///
+/// Opens the directory with `read` permission using the [`open_logic`] flow, then calls
+/// [`fdopendir`] to convert the [`RawFd`] into a `*DIR` stream (which we treat as `usize`).
+#[hook_guard_fn]
+pub(super) unsafe extern "C" fn opendir_detour(raw_filename: *const c_char) -> usize {
+    open_logic(raw_filename, O_RDONLY, 0)
```

Comment:
need to handle case open succeeds but fdopendir fails (close fd)

Reply:
I'd add O_DIRECTORY just to be on the safe side (though we ignore it, but maybe we wont in the future)